chain-impl-mockchain/src/header/header.rs

```diff
@@ -63,6 +63,36 @@ impl HeaderGenesisPraos {
     }
 }
 
+/// Header description
+#[derive(Clone)]
+pub struct HeaderDesc {
+    pub id: HeaderId,
+    pub date: BlockDate,
+    pub height: ChainLength,
+}
+
+impl fmt::Debug for HeaderDesc {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}-{:08x}-{}.{}",
+            self.id, self.height.0, self.date.epoch, self.date.slot_id,
+        )
```

Comment:
Why not derive `Debug` in the conventional way? It's `Display` that should work as the compact human-readable representation.

Reply:
I think this could be seen as a primitive which make sense to have a more compact impl than the normal `Debug`. Also it's mostly "historical" is there as the previous debug implementation before committing was shorting the id to 16 characters